### ISSUE TITLE
Include init_score on the Python Booster class

### ISSUE THREAD

**Issue by JoshuaC3** [microsoft/LightGBM#4065]
## Summary

Add `init_score_values` as a parameter on the Python [Booster](https://github.com/microsoft/LightGBM/blob/master/python-package/lightgbm/basic.py#L2158) class. Currently, it is only available on the [Dataset](https://github.com/microsoft/LightGBM/blob/master/python-package/lightgbm/basic.py#L1051) class.

## Motivation

Currently, it is a little tricky to find `init_score` after a model is trained. While it can be set, and therefore known, when [`boost_from_average`](https://lightgbm.readthedocs.io/en/latest/Parameters.html#boost_from_average) is isn't. When saving a model, the easy access to this presumably gets lost. Therefore, it would be useful to have it as a parameter on the model.

It then allows us to access it via the sklearn API.

## Description

See [here](https://github.com/microsoft/LightGBM/issues/3905) for further details and initial motivation.

**Comment by JoshuaC3** [comment]
Possibly something like,

```
    for dataset_name, eval_name, score, _ in evaluation_result_list:
        booster.best_score[dataset_name][eval_name] = score

    booster.init_score_value = trainset.get_init_score()

    if not keep_training_booster:
        booster.model_from_string(booster.model_to_string(), False).free_dataset()
```

[here](https://github.com/microsoft/LightGBM/blob/master/python-package/lightgbm/engine.py#L269).

**Comment by StrikerRUS** [comment]
Closed in favor of being in #2302. We decided to keep all feature requests in one place.

Welcome to contribute this feature! Please re-open this issue (or post a comment if you are not a topic starter) if you are actively working on implementing this feature.
